packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -50,30 +50,48 @@ struct UnresolvedWitnessIndexes {
 #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
 pub trait BuildableTransaction: sealed::Sealed {
     type TxType: Transaction;
+    type Context;
 
-    async fn build(self, provider: impl DryRunner) -> Result<Self::TxType>;
+    async fn build(self, provider: impl DryRunner, context: Self::Context) -> Result<Self::TxType>;
+}
+
+impl sealed::Sealed for ScriptTransactionBuilder {}
 
+#[derive(Debug, Clone, Default)]
+pub enum ScriptContext {
+    /// Transaction is estimated and signatures are automatically added.
+    #[default]
+    Normal,
+    /// Transaction is estimated but no signatures are added.
     /// Building without signatures will set the witness indexes of signed coins in the
     /// order as they appear in the inputs. Multiple coins with the same owner will have
     /// the same witness index. Make sure you sign the built transaction in the expected order.
-    async fn build_without_signatures(self, provider: impl DryRunner) -> Result<Self::TxType>;
+    NoSignatures,
```

Comment:
This might be problematic when witness data is used for something other than signatures. The assumed indexes are most likely going to be wrong. 
It's not a bug per se but an easy footgun.

Reply:
Good catch. Currently we ignore already added witnesses. I will fix this in this PR an then the comment will be correct.

Reply:
My previous comment was wrong. Manually added witnesses were considered.